nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py

```diff
@@ -361,8 +363,23 @@ def get_ref_policy_logprobs(self, batch):
         pos_ids = torch.cat((batch["position_ids"], batch["position_ids"]), dim=0)
         labels = torch.cat((batch["chosen_labels"], batch["rejected_labels"]), dim=0)
         global_batch = [tokens, masks, pos_ids, labels]
-        with cpu_weight_swap(self, self.ref_policy_state_dict, megatron_amp_O2=self.megatron_amp_O2):
+
+        if self.use_peft and self.ref_policy_state_dict is None:
+            # disable adapters -> reference model
+            for _, module in self.named_modules():
+                if isinstance(module, AdapterModuleMixin) and module.is_adapter_available():
+                    module.set_enabled_adapters(enabled=False)
+
             ref_log_probs = self.get_logprob_batch(global_batch)
```

Comment:
Is the goal here to calculate the log probs from the frozen model weights, whilst you use the LoRA weights for non-reference log probs?

Reply:
when using lora instead of full-tuning, the actor is reference model + lora. Therefore, we can simply disable lora (don't pass through it during forward pass) to get the reference logprobs

Reply:
How can the actor be reference + lora if the reference model is always None in `train_gpt_dpo.py`?

Reply:
Oh I mean:
for full dpo, we have reference model(in ```ref_policy_state_dict```), actor(initialized by reference and fully trainable)
for lora dpo, we don't have reference model in ```ref_policy_state_dict```. However the actor is initialized by reference+lora, and only the lora part is trained. So when we want to get ref_log_prob, we take out the lora part and use the main skeleton in actor (which is reference)

Reply:
Okay, please update the documentation to reflect that this is how DPO + LoRA works

Reply:
Document is updated https://github.com/NVIDIA/NeMo-Aligner/pull/142/commits/a4a511cb0d79a8fb98025aa59c1f525f16c89cc7